zvariant/src/file_path.rs

```diff
@@ -0,0 +1,106 @@
+use std::{ffi::{OsStr, OsString}, path::{Path, PathBuf}};
+
+use serde::{de::{self, Visitor}, Deserialize, Deserializer, Serialize, Serializer};
+use crate::de::Deserializer as ZbusDeserializer;
+
+use crate::Type;
+
+/// A file name represented as a nul-terminated byte array.
```

Comment:
Could you please add a bit more description on second paragraph:

```
While `zvariant::Type` and `serde::{Serialize, Deserialize}`, are implemented for [`Path`] and [`PathBuf`], unfortunately `serde` serializes them as UTF-8 strings. This is not the desired behavior in most cases since file paths are not guaranteed to only contain UTF-8 characters.

To solve this problem, this type is provided which encodes the underlying file path as a null-terminated byte array. Encoding as byte array is also more efficient.
```